src/consumer/mod.rs

```diff
@@ -12,8 +12,56 @@ const STATUS_RUNNING: usize = 0;
 const STATUS_QUIET: usize = 1;
 const STATUS_TERMINATING: usize = 2;
 
-type JobRunner<E> = dyn Fn(Job) -> Result<(), E> + Send + Sync;
-type BoxedJobRunner<E> = Box<JobRunner<E>>;
+/// Implementors of this trait can be registered to run jobs in a `Consumer`.
+///
+/// # Example
+///
+/// Create a worker with all default options, register a single handler (for the `foo` job
+/// type), connect to the Faktory server, and start accepting jobs.
+/// The handler is a struct that implements `JobRunner`.
+///
+/// ```no_run
+/// use faktory::{ConsumerBuilder, JobRunner, Job};
+/// use std::io;
+///
+/// struct MyHandler {
+///     config: String,
+/// }
+/// impl JobRunner<io::Error> for MyHandler {
+///    fn run(&self, job: Job) -> Result<(), io::Error> {
+///       println!("config: {}", self.config);
+///       println!("job: {:?}", job);
+///       Ok(())
+///   }
+/// }
+///
+/// let mut c = ConsumerBuilder::default();
+/// let handler = MyHandler {
+///    config: "bar".to_string(),
+/// };
+/// c.register_runner("foo", Box::new(handler));
+/// let mut c = c.connect(None).unwrap();
+/// if let Err(e) = c.run(&["default"]) {
+///     println!("worker failed: {}", e);
+/// }
+/// ```
+pub trait JobRunner: Send + Sync {
+    /// The error type that the handler may return.
+    type E;
+    /// A handler function that runs a job.
+    fn run(&self, job: Job) -> Result<(), Self::E>;
```

Comment:
I haven't quite figured out this yet, I think I need to understand the rules about overlapping blanket implementations more closely

Reply:
I think I have this done, the `BoxRunner` trait is implemented for `Box<F>`, `&F`, and `&mut F`,
where `F: Fn(Job) -> Result<(), E> + Send + Sync`.
It is no longer implemented for `F` because it appears to overlap with all the other implementations.